### PR TITLE
[Temporary suspension. Expected to resume before 2021.08.24] Monika's chess tutorial

### PR DESCRIPTION
**Pull request by Wingdinggaster656**
It took me a while to write this.
I haven't really done it yet(just a few more lessons), but I think it's going to take a while to go through these files, so let me do it while you guys go through it.
I changed some of the chess.rpy files and added some functions to get better highlights. The puzzle system has also been added. Though puzzle system's ELO system is not really done.
Anyways, the following are this PR's content:

### Chess Lessons(script-chess-lesson.rpy).
The longest section. Contains a number of tutorials mainly for beginners, but also somewhat benefits advancer.
Current lesson list:
- [x] Can you teach me about chess?
- [x] Terms in chess
- [ ] Chess - Basic rule
- [x] Introduction to pieces.
- [x] Advanced introduction to pieces.
- [x] Some Openings.
- [x] Key Point 1 - ELO rating
- [x] Key Point 2 - PGN standard
- [x] Basic endgame patterns.
- [x] Baisc tactics.

### Chess Points(script-chess-points.rpy)
All kinds of chess knowledge are included here.
It's also a "fun fact list" in a sense. Personally, I prefer this as a supplement and extension to the lesson.'
Not really done, 4 points waiting to be added.

### Chess Puzzles(script-chess-puzzle.rpy)
It is possible that this system will not be added in this release.
Chess puzzles are also an extremely important part of improving chess, so I added them. To be completed.

**To sum up: I suggest that for the time being to mainly check the script-chess-lesson file, and lower the priority of the other two files, because the other two files are my main unfinished business at the moment.**

### CODE REVIEW

File: Monika After Story/game/script-chess-lesson.rpy [hunk 1]
```diff
@@ -0,0 +1,5325 @@
...
+    m 3eua "Remember that chess lesson I said I was going to give you?"
+    m 3hua "I've got it all planned out!"
+    m 3husdla "Although they are a bit messy...{w=0.4}{nw}"
+    extend 3husdlb " and I don't have a strict order for them either."
+    if persistent._mas_pm_player_chesslevel == chesslevel_master:
+        m 3rusdlb "And considering that you're already a master at chess, I'm afraid these lessons seem rather useless to you..."
+    elif persistent._mas_pm_player_chesslevel == chesslevel_advancer:
+        m 3rusdlb "And considering that you say you've done quite a bit of work on chess, I hope you'll tolerate some of the more verbose aspects of my lesson."
+    m 1eua "But anyway, if you see any lessons that interest you, just ask me."
+    m 1esb "There are also some courses that I won't offer you right now...{w=0.2} I'm only going to talk to you about those advanced stuffs after you've taken certain classes."
```
Comment (fromulus) [inline Monika After Story/game/script-chess-lesson.rpy:230]:
```suggestion
    m 1esd "There are also some things that I won't teach you just yet... {w=0.3}{nw}"
    extend 1esa "I think it makes more sense to keep the more advanced stuff for later."
```

File: Monika After Story/game/script-chess-lesson.rpy [hunk 2]
```diff
@@ -0,0 +1,5325 @@
...
+    m 1hub "That's about all I have to say. If you want to have a class, you can start now!"
```
Comment (fromulus) [inline Monika After Story/game/script-chess-lesson.rpy:231]:
```suggestion
    m 1hua "That's about all I had to say. {w=0.3}{nw}"
    extend 1hub "I can start teaching you whenever you want, just ask~"
```

**Review by fromulus — COMMENTED** [review]
Disclaimer : I don't know sh*t about how to play chess, I'm just going to review dialogue lines that sound weird to me, disregard my changes in case there are some chess-related expressions I don't know about.

I'm going to do this in multiple reviews due to how gigantic this PR is.

**Comment by Wingdinggaster656** [comment]
@fromulus
Thank you so much for taking the time to check these out!
Then, it's true there are some parts of the problem that you're pointing out that are "chess-way expressions". but it's also true that there are some parts of the conversation are better with your words. I will correct them at a later date!
Again, thanks for helping me review the dialogue!

**Comment by multimokia** [comment]
For the sake of being able to review this with more ease, can we split each of these topics into their own PR?

Keep the base framework they all use (if any additional code was involved) with the first topic, which can be kept in this PR, then the rest of the topics can be merged in more easily with reviews.

it's just much harder to review a PR that's this large (7k+ lines)

**Comment by Wingdinggaster656** [comment]
I think there's no problem at all. I'll do it.

There's just one small trouble: I made these chess lessons before I was busy with my other work.
In other words, I will be too busy with other work issues to deal with this for the time being.
I think I'll have free time in about two weeks to re-establish those PR, which is saying I'm afraid I won't be able to deal with these things now.

Conclusion: I'll do it, but it's about two weeks later.

**Comment by Wingdinggaster656** [comment]
Just writing this to let you know I'm not dead. I'm simply a little busy and can't get away from it right now.
Rest assured, I'll finish this, after all, I've written thousands of lines, only the last sort part remains.

**Comment by Wingdinggaster656** [comment]
This PR will not be abandoned, as I said before.
However, in order to improve the overall chess system (of course, I won't let it take the lead, after all, the mod is not called Monika Chess Story), I will look at this after my `ChessButHumanized` project(one of them:#9117) is completed, and then Monika can teach it via ELO, also implement many other functions.

[Close temporarily to wait for other processes to complete]
